### PR TITLE
Adding OTP functionality

### PR DESCRIPTION
**Pull request by cedric06nice**
Adding functionality to allow users with OTP/MFA enabled to login to the API.

### CODE REVIEW

File: lib/src/idirectus_api_manager.dart
Comment (maxbritto) [inline lib/src/idirectus_api_manager.dart:1]: Maybe it would be simpler to add an optional named parameter `oneTimePassword` in the existing login method instead of creating another method ?

File: pubspec.yaml
Comment (maxbritto) [inline pubspec.yaml:1]: Doesn’t the lint upgrade leads to new warnings ? If that is the case we should not include it in the merge, and do it later with the fix included

File: test/directus_api_test.dart
Comment (maxbritto) [inline test/directus_api_test.dart:1]: You also updated the `test('Get list of items with filter that includes special characters', () { ` but I do not see the link with the rest of your patch. 
if this is a real problem that need fixing, it should be on its own patch, merge request and commit message.

**Review by maxbritto — CHANGES_REQUESTED** [review]
Thanks @cedric06nice for this patch, this is a nice addition!
I made some comments for specific small issues. 
Let me know if you need help doing those changes.

File: lib/src/idirectus_api_manager.dart
Comment (cedric06nice) [reply lib/src/idirectus_api_manager.dart:1]: Totally make sense! Code reviewed to reflect one only method.

File: pubspec.yaml
Comment (cedric06nice) [reply pubspec.yaml:1]: I havent noticed any but downgraded it for this merge.

File: test/directus_api_test.dart
Comment (cedric06nice) [reply test/directus_api_test.dart:1]: It leads to an error for me but cancelled the change as unrelated to OTP functionality.

File: lib/src/directus_api_manager_base.dart
Comment (maxbritto) [inline lib/src/directus_api_manager_base.dart:1]: Maybe it would be nicer and more "darter" to use named parameters for optional parameters. It will increase discoverability for this new feature as it will show up on auto-complete.

File: lib/src/idirectus_api_manager.dart [hunk 1]
```diff
@@ -4,7 +4,7 @@ import 'package:http/http.dart';
 
 abstract class IDirectusApiManager {
   Future<DirectusLoginResult> loginDirectusUser(
-      String username, String password);
+      String username, String password, String? oneTimePassword);
```
Comment (maxbritto) [inline lib/src/idirectus_api_manager.dart:7]: This is not the same signature as the implementation where you set the parameter as optional and here it is mandatory. Using named parameter should help as it won't be building with differences like this.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 name: directus_api_manager
 description: Communicate with a Directus server using its REST API.
 version: 1.7.2
```
Comment (maxbritto) [inline pubspec.yaml:3]: You should increase the version number to 1.7.3 and also update the changelog.md to document this new feature

**Review by maxbritto — CHANGES_REQUESTED** [review]
Way better like this, thanks!
Still added some feedback regarding the optional parameter and also the need to increase the version number and updating the changeling. 
After that we should be good to go :)

File: lib/src/directus_api.dart [hunk 3]
```diff
@@ -71,9 +71,7 @@ abstract class IDirectusAPI {
   BaseRequest? prepareLogoutRequest();
   bool parseLogoutResponse(Response response);
 
-  Request prepareLoginRequest(String username, String password);
-  Request prepareLoginRequestWithOtp(
-      String username, String password, String otp);
+  Request prepareLoginRequest(String username, String password, String? oneTimePassword);
```
Comment (maxbritto) [inline lib/src/directus_api.dart:74]: Sorry, misread the previous patch request, here is the most important place for the `oneTimePassword` to be made as a named parameter : `{String? oneTimePassword}`. It's public facing, hence should be as clear as possible. That being said, the `prepareLoginRequestMethod` would be improved too by using a named parameter as well.

File: pubspec.yaml (hunk 2 above)
Comment (cedric06nice) [reply pubspec.yaml:3]: Increasing to 1.7.3 the version and added to the changelog.md file. We are missing 1.7.2 changes in the changelog. Should it not be version 1.8.0 as a 'new feature with no breaking changes' (even if this is a small new feature)?

File: lib/src/idirectus_api_manager.dart (hunk 1 above)
Comment (cedric06nice) [reply lib/src/idirectus_api_manager.dart:7]: Changed from unnamed optional [String? oneTimePassword] to named optional {String? oneTimePassword}.

File: lib/src/directus_api_manager_base.dart
Comment (cedric06nice) [reply lib/src/directus_api_manager_base.dart:1]: Great idea!

File: lib/src/directus_api.dart (hunk 3 above)
Comment (cedric06nice) [reply lib/src/directus_api.dart:74]: Changed from unnamed optional [String? oneTimePassword] to named optional {String? oneTimePassword}.

File: pubspec.yaml (hunk 2 above)
Comment (maxbritto) [reply pubspec.yaml]: Good catch, 1.8.0 is better in this case.
My bad on 1.7.2 changelog missing, I'll add it after your commit to spare you a git conflict

Comment (cedric06nice) [reply pubspec.yaml]: I have tagged it v1.7.3 now, but next time I will remember the convention.
